common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -1154,9 +1154,33 @@ public static StringSearch getStringSearch(
    * Returns the collation ID for the given collation name.
    */
   public static int collationNameToId(String collationName) throws SparkException {
+    // If collation name is given as a fully qualified name, extract the actual collation name as
+    // the last part of the [catalog].[schema].[collation_name] name.
+    long numDots = collationName.chars().filter(ch -> ch == '.').count();
```

Comment:
How about '.' in backticks like 
```
`catalog.name`.schema.collation_name
```

Reply:
Don't we have standard functions for this. Also who would use this method in this way?
If teh collation was parsed it should be passed as a Seq of identifiers (or something like it). It should NOT have been assembled as a qualified string.
At this point we can also just "eaT" teh qualifiers after chjecking that they are correct.
only once we have user defined collatiosn will we need to actually deal with non SYSTEM.BUILTIN qualfiers.
That day may not come for years.

Reply:
Can we leverage `AttributeNameParser.parseAttributeName`?

Reply:
`AttributeNameParser.parseAttributeName` would introduce a circular dependency. Do we have something similar in `spark/common`? Can i move this trait - @HyukjinKwon ?

Reply:
Please, put the parsing code to a common place, and reuse it.

Reply:
Do you have a place in mind?

Reply:
common/utils?

Reply:
`AttributeNameParser` has a dependency to `org.apache.spark.sql.errors.DataTypeErrors` which is in spark/sql, which then itself depends on other error classes also in sql.

If we want to remove this dependency then we need to introduce new error class for `AttributeNameParser` which would be a breaking change?

Reply:
Can you at least copy and paste, and add a pointer or something? Or you can make it inherited etc. This manual different parsing logic should be got rid of in any event.

Reply:
Added `AttributeNameParser.parseAttributeName` to resolve collation name. Refactored the code so the parsing happens in the analyzer as @cloud-fan suggested.